src/api/hooks/useWeatherApi.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useState } from 'react';
+import { City, fetchWeatherData } from '../weather';
+
+export const useWeatherApi = () => {
+    const [cities, setCities] = useState<City[]>([]);
+    const [loading, setLoading] = useState<boolean>(true);
+    const [error, setError] = useState<string | null>(null);
+
+    useEffect(()=> {
+    fetchWeatherData().then(data => {
```

Comment:
Looks like this is still using chaining rather than async await.

Here are some good resources on it:
 - [Chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises#chaining)
 - [Async/Await](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Promises#async_and_await)